Testsuite fails in Debian after updating to 10.0.x branch

I am currently trying to update the Debian package from `9.25.22` to `10.0.15` and so far have been unsuccessful to build the updated version.

So far, I made the following changes to the packaging source:

```
diff --git a/debian/control b/debian/control
index 9d5d69b..a4e7073 100644
--- a/debian/control
+++ b/debian/control
@@ -6,11 +6,13 @@ Uploaders: Marcus Schäfer <marcus.schaefer@gmail.com>
 Build-Depends: debhelper-compat (= 13),
                dh-python,
                passwd,
+               pybuild-plugin-pyproject,
                python3-all,
                python3-anymarkup,
                python3-docopt,
                python3-lxml,
                python3-mock,
+               python3-poetry,
                python3-pytest,
                python3-pyxattr,
                python3-requests,
diff --git a/debian/rules b/debian/rules
index 437e959..dbe59d7 100755
--- a/debian/rules
+++ b/debian/rules
@@ -20,7 +20,7 @@ ifeq ($(filter nocheck nobench, $(DEB_BUILD_OPTIONS)),)
 endif
 
 override_dh_auto_build:
-       make tools
+#      make tools
        dh_auto_build
        make -C doc man
 
```

The build fails with:

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tasks/base_test.py::TestCliTask::test_load_xml_description - kiwi.exce...
FAILED tasks/base_test.py::TestCliTask::test_load_xml_description_buildservice
ERROR cli_test.py::TestCli::test_show_and_exit_on_help_request - kiwi.excepti...
ERROR cli_test.py::TestCli::test_get_servicename_system - kiwi.exceptions.Kiw...
ERROR cli_test.py::TestCli::test_warning_on_use_of_legacy_disk_type - kiwi.ex...
ERROR cli_test.py::TestCli::test_set_target_arch - kiwi.exceptions.KiwiComman...
ERROR cli_test.py::TestCli::test_get_servicename_image - kiwi.exceptions.Kiwi...
ERROR cli_test.py::TestCli::test_get_servicename_result - kiwi.exceptions.Kiw...
ERROR cli_test.py::TestCli::test_get_command - kiwi.exceptions.KiwiCommandNot...
ERROR cli_test.py::TestCli::test_get_command_args - kiwi.exceptions.KiwiComma...
ERROR cli_test.py::TestCli::test_load_command - kiwi.exceptions.KiwiCommandNo...
ERROR cli_test.py::TestCli::test_load_command_unknown - kiwi.exceptions.KiwiC...
ERROR cli_test.py::TestCli::test_load_command_undefined - kiwi.exceptions.Kiw...
ERROR cli_test.py::TestCli::test_get_command_args_not_loaded - kiwi.exception...
ERROR cli_test.py::TestCli::test_get_servicename_unknown - kiwi.exceptions.Ki...
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info - ...
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_color_output
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_resolve_package_list
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_add_repo
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_delete_repos
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_help
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_print_xml
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_print_yaml
ERROR tasks/image_info_test.py::TestImageInfoTask::test_process_image_info_print_toml
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_no_root_directory_specified
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_no_raw_disk_found
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_unsupported_size_format
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_image_resize_gb
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_image_resize_mb
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_image_resize_bytes
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_image_resize_not_needed
ERROR tasks/image_resize_test.py::TestImageResizeTask::test_process_image_resize_help
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_invalid_bundle_directory
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_result_bundle
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_result_bundle_as_rpm
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_result_bundle_with_bundle_format
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_result_bundle_with_bundle_format_from_commandline
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_result_bundle_name_includes_version
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_result_bundle_zsyncmake_missing
ERROR tasks/result_bundle_test.py::TestResultBundleTask::test_process_result_bundle_help
ERROR tasks/result_list_test.py::TestResultListTask::test_process_result_list
ERROR tasks/result_list_test.py::TestResultListTask::test_process_result_list_help
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_add_package
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_update_delete_package
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_prepare_stage_set_container_tag
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_add_container_label
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_add_container_label_invalid_format
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_prepare_stage_set_derived_from_uri
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_prepare_stage_set_repo
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_prepare_stage_add_repo
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_build_help
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_prepare_ignore_repos
ERROR tasks/system_build_test.py::TestSystemBuildTask::test_process_system_prepare_ignore_repos_used_for_build
ERROR tasks/system_create_test.py::TestSystemCreateTask::test_process_system_create
ERROR tasks/system_create_test.py::TestSystemCreateTask::test_process_system_create_help
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_run_debootstrap_only_once
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_add_package
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_delete_package
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_set_container_tag
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_add_container_label
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_add_container_label_invalid_format
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_set_derived_from_uri
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_set_repo
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_add_repo
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_help
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_delete_repos
ERROR tasks/system_prepare_test.py::TestSystemPrepareTask::test_process_system_prepare_delete_repos_used_for_build
ERROR tasks/system_update_test.py::TestSystemUpdateTask::test_process_system_update
ERROR tasks/system_update_test.py::TestSystemUpdateTask::test_process_system_update_add_package
ERROR tasks/system_update_test.py::TestSystemUpdateTask::test_process_system_update_delete_package
ERROR tasks/system_update_test.py::TestSystemUpdateTask::test_process_system_update_help
======= 2 failed, 1625 passed, 1 warning, 71 errors in 442.62s (0:07:22) =======
```

I assume I'm missing build dependencies. However, it's not very obvious which are missing.

Would be great to get some advise, so I can speedily update the package and fix the outstanding FTBFS bug [1].

> [1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1069389

--- a/kiwi/cli.py
+++ b/kiwi/cli.py
@@ -119,7 +119,7 @@ global options for services: image, system
 import logging
 import sys
 import os
-import pkg_resources
+from importlib.metadata import entry_points
 from docopt import docopt
 
 # project
@@ -252,10 +252,16 @@ class Cli:
 
         :rtype: object
         """
-        discovered_tasks = {
-            entry_point.name: entry_point.load()
-            for entry_point in pkg_resources.iter_entry_points('kiwi.tasks')
-        }
+        discovered_tasks = {}
+        if sys.version_info >= (3, 12):
+            for entry in list(entry_points()):  # pragma: no cover
+                if entry.group == 'kiwi.tasks':
+                    discovered_tasks[entry.name] = entry.load()
+        else:  # pragma: no cover
+            module_entries = dict.get(entry_points(), 'kiwi.tasks')
+            for entry in module_entries:
+                discovered_tasks[entry.name] = entry.load()
+
         service = self.get_servicename()
         command = self.get_command()
 

--- a/kiwi/defaults.py
+++ b/kiwi/defaults.py
@@ -18,12 +18,13 @@
 import logging
 import os
 import glob
+import importlib
+from importlib.resources import as_file
 from importlib.util import find_spec
 from importlib.machinery import ModuleSpec
 from collections import namedtuple
 import platform
 import yaml
-from pkg_resources import resource_filename
 from typing import (
     List, NamedTuple, Optional, Dict
 )
@@ -1682,11 +1683,22 @@ class Defaults:
         return Defaults.project_file('xsl/master.xsl')
 
     @staticmethod
+    def get_schematron_module_name():
+        """
+        Provides module name for XML SchemaTron validations
+
+        :return: python module name
+
+        :rtype: str
+        """
+        return 'lxml.isoschematron'
+
+    @staticmethod
     def project_file(filename):
         """
         Provides the python module base directory search path
 
-        The method uses the resource_filename method to identify
+        The method uses the importlib.resources.path method to identify
         files and directories from the application
 
         :param string filename: relative project file
@@ -1695,7 +1707,8 @@ class Defaults:
 
         :rtype: str
         """
-        return resource_filename('kiwi', filename)
+        with as_file(importlib.resources.files('kiwi')) as path:
+            return f'{path}/{filename}'
 
     @staticmethod
     def get_imported_root_image(root_dir):

--- a/kiwi/xml_description.py
+++ b/kiwi/xml_description.py
@@ -83,7 +83,9 @@ class XMLDescription:
         isoschematron = None
         schematron = None
         try:
-            isoschematron = importlib.import_module('lxml.isoschematron')
+            isoschematron = importlib.import_module(
+                Defaults.get_schematron_module_name()
+            )
         except Exception as error:
             log.warning(f"schematron validation skipped: {error}")
         try:

--- a/test/unit/xml_description_test.py
+++ b/test/unit/xml_description_test.py
@@ -172,11 +172,11 @@ class TestSchema:
         with raises(KiwiSchemaImportError):
             self.description_from_file.load()
 
-    @patch('importlib.import_module')
+    @patch('kiwi.xml_description.Defaults.get_schematron_module_name')
     def test_load_schema_from_xml_content_skipping_isoschematron(
-        self, mock_import_module
+        self, mock_get_schematron_module_name
     ):
-        mock_import_module.side_effect = Exception
+        mock_get_schematron_module_name.return_value = 'bogus'
         with self._caplog.at_level(logging.WARNING):
             self.description_from_data.load()
             assert 'schematron validation skipped:' in self._caplog.text

